File: airbyte-integrations/connectors/destination-postgres/src/test-integration/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresRawOverrideDisableTypingDedupingTest.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.postgres.typing_deduping;

import com.fasterxml.jackson.databind.node.ObjectNode;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

public class PostgresRawOverrideDisableTypingDedupingTest extends PostgresTypingDedupingTest {

  @Override
  protected ObjectNode getBaseConfig() {
    return super.getBaseConfig()
        .put("raw_data_schema", "overridden_raw_dataset")
        .put("disable_type_dedupe", true);
  }

  @Override
  protected String getRawSchema() {
    return "overridden_raw_dataset";
  }

  @Override
  protected boolean disableFinalTableComparison() {
    return true;
  }

  @Disabled
  @Test
  @Override
  public void identicalNameSimultaneousSync() {}

  @Disabled
  @Test
  @Override
  public void testVarcharLimitOver64K() {}

}

File: airbyte-integrations/connectors/destination-postgres/src/test-integration/kotlin/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresRawOverrideDisableTypingDedupingTest.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.integrations.destination.postgres.typing_deduping

import com.fasterxml.jackson.databind.node.ObjectNode
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test

class PostgresRawOverrideDisableTypingDedupingTest : PostgresTypingDedupingTest() {
    override fun getBaseConfig(): ObjectNode {
        return super.getBaseConfig()
            .put("raw_data_schema", "overridden_raw_dataset")
            .put("disable_type_dedupe", true)
    }

    override val rawSchema: String
        get() = "overridden_raw_dataset"

    override fun disableFinalTableComparison(): Boolean {
        return true
    }

    @Disabled @Test override fun identicalNameSimultaneousSync() {}

    @Disabled @Test override fun testVarcharLimitOver64K() {}
}
